### ISSUE TITLE
TOC component should use a flexible implementation to retrieve headlines

### ISSUE THREAD

**Issue by tiberiuichim** [plone/volto#1616]
I think it would be interesting to see a more flexible implementation of the TOC component. It should allow each block to decide if it has headlines, and which headlines it has.

I propose that the ``blocksConfig`` object for each type has a new key, "extractHeadlines" (or maybe extractTOC?), which can return a list of headlines (or a single headline?) that block contains.

Let's discuss this.

**Comment by pnicolli** [comment]
@tiberiuichim this is solved now with `tocEntry` in the blocks config, right?

**Comment by davisagli** [comment]
@pnicolli Yes and no. `tocEntry` allows a block to provide a single headline but not a list.

**Comment by erral** [comment]
And even the headline level detection thing should be more flexible, right now it uses the second character of the style name
